src/IndexingMsg.chpl

```diff
@@ -13,9 +13,166 @@ module IndexingMsg
 
     use CommAggregation;
 
+    use FileIO;
+    use List;
+
     private config const logLevel = ServerConfig.logLevel;
     const imLogger = new Logger(logLevel);
 
+    proc jsonToTuple(json: string, type t) throws {
+        var f = opentmp(); defer { ensureClose(f); }
+        var w = f.writer();
+        w.write(json);
+        w.close();
+        var r = f.reader(start=0);
+        var tup: t;
+        r.readf("%jt", tup);
+        r.close();
+        return tup;
+    }
+
+    proc arrayViewMixedIndexMsg(cmd: string, payload: string, st: borrowed SymTab): MsgTuple throws {
+        var (pdaName, indexDimName, ndimStr, dimProdName, coords) = payload.splitMsgToTuple(5);
+        var ndim = try! ndimStr:int;
+        imLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+                                                    "%s %s %i %s %s".format(cmd, pdaName, ndim, dimProdName, coords));
+
+        var dimProd: borrowed GenSymEntry = getGenericTypedArrayEntry(dimProdName, st);
+        var dimProdEntry = toSymEntry(dimProd, int);
+
+        var indexDim: borrowed GenSymEntry = getGenericTypedArrayEntry(indexDimName, st);
+        var indexDimEntry = toSymEntry(indexDim, int);
+        ref dims = indexDimEntry.a;
+
+        // String array containing the type of the following value at even indicies then the value ex. ["int", "7", "slice", "(0,5,-1)", "pdarray", "id_4"]
+        var typeCoords: [0..#(ndim*2)] string = jsonToPdArray(coords, ndim*2);
+
+        var scaledCoords: [makeDistDom(+ reduce dims)] int;
+        // check there's enough room to create a copy for scan and throw if creating a copy would go over memory limit
+        overMemLimit(numBytes(int) * dims.size);
+        var offsets = (+ scan dims) - dims;
+
+        for i in 0..#typeCoords.size by 2 {
+            select typeCoords[i] {
+                when "int" {
+                    scaledCoords[offsets[i/2]] = typeCoords[i+1]:int * dimProdEntry.a[i/2];
+                }
+                when "slice" {
+                    var (start, stop, stride) = jsonToTuple(typeCoords[i+1], 3*int);
+                    var slice: range(stridable=true) = convertSlice(start, stop, stride);
+                    var scaled: [0..#slice.size] int = slice * dimProdEntry.a[i/2];
+                    for j in 0..#slice.size {
+                        scaledCoords[offsets[i/2]+j] = scaled[j];
+                    }
```

Comment:
I don't totally understand the calculation that is going on here, but I think that since `scaledCoords` is a distributed array, there are going to be some remote accesses. 

Since `dimProdEntry` is also a distributed array, there will also be some remote accesses there, which make things a little more difficult than just doing a `coforall` over `scaledCoords` local subdomains. 

This may be an unhelpful comment since I don't really know what a solution would be, but I think some better locality could be achieved potentially. @ronawho I'm curious if you have any input here.